pkg/graph/crd/compat/compat.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2025 The Kube Resource Orchestrator Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package compat
+
+import (
+	"fmt"
+
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+)
+
+// DiffSchema compares schema versions and returns compatibility details.
+// This function expects exactly one version in each slice. If more versions are present,
+// it will return an error as multi-version support is not implemented.
+func DiffSchema(oldVersions []v1.CustomResourceDefinitionVersion, newVersions []v1.CustomResourceDefinitionVersion) (*DiffResult, error) {
```

Comment:
Looking at https://github.com/kubernetes/kubernetes/blob/dd43c3d29d5203378ce456de9466597234a83e66/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go#L178

- removal of sub-resources might matter?
- flopping Deprecated could be bad
- setting served or storage to false would break KRO